rust/lance-index/src/scalar.rs

```diff
@@ -98,6 +98,40 @@ impl PartialEq for dyn AnyQuery {
     }
 }
 
+/// A full text search query
+#[derive(Debug, Clone, PartialEq)]
+pub struct FullTextSearchQuery {
+    /// The columns to search,
+    /// if empty, search all indexed columns
+    pub columns: Vec<String>,
+    /// The full text search query
+    pub query: String,
+    /// The maximum number of results to return
+    pub limit: Option<i64>,
```

Comment:
I don't think we have any python bindings for `limit` (which is ok if intentional).

Also,  how does the `limit` here compare to the `limit` that the scanner already has?  I think it might be confusing to have two limits.

Reply:
This is used to limit the number of results the index returns for performance, it's not exposed and will always be set to the `limit` that scanner has.
For what you mentioned that we should have a FTS plan node, i think this would be removed once we have that.